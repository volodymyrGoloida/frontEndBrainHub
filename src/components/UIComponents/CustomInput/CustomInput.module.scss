@import "../../../UI-global-styles/_variables.scss";
@import "../../../UI-global-styles/mixins";

.root {
  @include DisplayFlexMixin(column, center, center);
  position: relative;
  input {
    padding: 0.7rem 1rem;
    box-sizing: border-box;
    width: 100%;
    background-color: $purple;
    border: none;
    color: white;
    &::placeholder {
      color: white;
    }
    &:focus {
      border: none;
      background-color: $white-purple;
    }
  }

  .inputError {
    font-size: 0.75rem;
    color: $danger;
    margin: 0.1rem 0.6rem 0;
  }
}
