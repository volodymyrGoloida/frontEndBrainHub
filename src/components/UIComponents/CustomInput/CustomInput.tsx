import React from "react";
import scss from "./CustomInput.module.scss";

interface CustomInputProps {
  error?: string;
  onChange: (name: string, value: string) => void;
  name: string;
  type: string;
  placeholder?: string;
  value?: string;
}

const CustomInput: React.FC<CustomInputProps> = ({
  error,
  onChange,
  name,
  type,
  placeholder = "",
  value,
}) => (
  <div className={scss.root}>
    <input
      onChange={(e) => onChange(e.target.name, e.target.value)}
      autoComplete="off"
      name={name}
      value={value}
      type={type}
      placeholder={placeholder}
    />
    {error && <p className={scss.inputError}>{error}</p>}
  </div>
);
export default CustomInput;
