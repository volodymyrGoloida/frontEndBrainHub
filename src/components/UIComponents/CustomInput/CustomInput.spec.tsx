import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import CustomInput from "./CustomInput";

describe("component CustomInput", () => {
  it("should successfully render CustomInput", () => {
    const { rerender, baseElement } = render(
      <CustomInput
        onChange={jest.fn}
        name="text"
        type="text"
        placeholder="please enter your email"
      />
    );
    const input = screen.getByPlaceholderText(
      "please enter your email"
    ) as HTMLInputElement;

    expect(input).toBeInTheDocument();
    fireEvent.change(input, {
      target: {
        value: "qqwe",
      },
    });

    expect(input.value).toBe("qqwe");

    rerender(
      <CustomInput
        onChange={jest.fn()}
        name="text"
        type="text"
        placeholder="please enter your email"
        error="qweqwe"
      />
    );
    const errorMessage = screen.getByText("qweqwe");
    expect(errorMessage).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
