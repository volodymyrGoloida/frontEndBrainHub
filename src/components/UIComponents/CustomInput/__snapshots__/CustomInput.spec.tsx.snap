// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component CustomInput should successfully render CustomInput 1`] = `
<body>
  <div>
    <div
      class="root"
    >
      <input
        autocomplete="off"
        name="text"
        placeholder="please enter your email"
        type="text"
        value=""
      />
      <p
        class="inputError"
      >
        qweqwe
      </p>
    </div>
  </div>
</body>
`;
